service-workers/service-worker/registration-updateviacache.https.html

```diff
@@ -0,0 +1,230 @@
+<!DOCTYPE html>
+<title>Service Worker: Registration-updateViaCache</title>
+<script src="/resources/testharness.js"></script>
+<script src="resources/testharness-helpers.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+  const updateViaCacheValues = [undefined, 'imports', 'all', 'none'];
+  const scriptUrl = 'resources/update-max-aged-worker.py';
+  const scope = 'resources/blank.html';
+
+  async function cleanup() {
+    const reg = await navigator.serviceWorker.getRegistration(scope);
+    if (!reg) return;
+    if (reg.scope == new URL(scope, location).href) {
+      return reg.unregister()
+    };
+  }
+
+  function wait(ms) {
+    return new Promise(r => setTimeout(r, ms));
+  }
+
+  function waitForActivated(sw) {
+    return new Promise((resolve, reject) => {
+      function check() {
+        if (sw.state == 'activated') {
+          sw.removeEventListener('statechange', check);
+          resolve();
+          return;
+        }
+        if (sw.state == 'redundant') {
+          sw.removeEventListener('statechange', check);
+          reject(Error('Service worker rejected'));
+          return;
+        }
+      }
+
+      sw.addEventListener('statechange', check);
+      check();
+    });
+  }
+
+  function getScriptTimes(sw, testName) {
+    return new Promise(resolve => {
+      navigator.serviceWorker.addEventListener('message', function listener(event) {
+        if (event.data.test !== testName) return;
+        navigator.serviceWorker.removeEventListener('message', listener);
+        resolve({
+          mainTime: event.data.mainTime,
+          importTime: event.data.importTime
+        });
+      });
+
+      sw.postMessage('');
+    });
+  }
+
+  function registerViaApi(scriptUrl, opts) {
+    return navigator.serviceWorker.register(scriptUrl, opts);
+  }
+
+  async function registerViaLinkElement(scriptUrl, opts) {
+    const link = document.createElement('link');
+
+    if (link.relList.supports('serviceworker') == false) throw Error("link rel=serviceworker not supported");
+
+    link.rel = 'serviceworker';
+    link.href = scriptUrl;
+    link.scope = opts.scope;
+
+    if (opts.updateViaCache) {
+      link.updateViaCache = opts.updateViaCache;
+    }
+
+    document.head.appendChild(link);
+
+    const fullScope = new URL(opts.scope, window.location).href;
+
+    while (true) {
+      const regs = await navigator.serviceWorker.getRegistrations();
+      const reg = regs.find(r => r.scope == fullScope && (r.installing || r.waiting || r.active));
+
+      if (reg) {
+        document.head.removeChild(link);
+        return reg;
+      }
+
+      await wait(100);
```

Comment:
The polling with timeout is a bit unfortunate, it puts a bound on how fast the test can run (100ms per test adds up for hundreds of tests). A similar test, register-link-element.https.html, just waits for link.onload. Could we just do that here?